src/devices/Common/Iot/Device/Common/WeatherHelper.cs

```diff
@@ -331,5 +346,84 @@ public static Length CalculateAltitude(Pressure pressure, Pressure seaLevelPress
+        {
+            var result = airPressure.Pascals / (SpecificGasConstantOfAir * temperature.Kelvins);
+            return Density.FromKilogramsPerCubicMeter(result);
+        }
+
+        /// <summary>
+        /// Calculates the air density
+        /// </summary>
+        /// <param name="airPressure">Measured air pressure</param>
+        /// <param name="temperature">Measured temperature</param>
+        /// <param name="humidity">Measured relative humidity</param>
+        /// <returns>Approximate standard air density at sea level</returns>
+        /// <remarks>From https://de.wikipedia.org/wiki/Luftdichte </remarks>
+        public static Density CalculateAirDensity(Pressure airPressure, Temperature temperature, RelativeHumidity humidity)
+        {
+            double rs = SpecificGasConstantOfAir;
+            double rd = SpecificGasConstantOfVapor;
+            var pd = CalculateSaturatedVaporPressureOverWater(temperature);
+            // It's still called "constant" even though it's not constant
+            double gasConstant = rs / (1 - (humidity.Percent / 100) * (pd.Pascals / airPressure.Pascals) * (1 - (rs / rd)));
+            var result = airPressure.Pascals / (gasConstant * temperature.Kelvins);
+            return Density.FromKilogramsPerCubicMeter(result);
+        }
+
+        /// <summary>
+        /// Calculates the wind chill temperature - this is the perceived temperature in (heavy) winds at cold temperatures.
+        /// This is only useful at temperatures below about 20°C, above use <see cref="CalculateHeatIndex"/> instead.
+        /// Not suitable for wind speeds &lt; 5 km/h.
+        /// </summary>
+        /// <param name="temperature">The measured air temperature</param>
+        /// <param name="windSpeed">The wind speed (measured at 10m above ground)</param>
+        /// <returns>The perceived temperature. Note that this is not a real temperature, and the skin will never really reach
+        /// this temperature. This is more an indication on how fast the skin will reach the air temperature. If the skin
+        /// reaches a temperature of about -5°C, frostbite might occur.</returns>
+        /// <remarks>From https://de.wikipedia.org/wiki/Windchill </remarks>
+        /// <exception cref="ArgumentOutOfRangeException">The wind speed is less than zero</exception>
+        public static Temperature CalculateWindchill(Temperature temperature, Speed windSpeed)
+        {
+            if (windSpeed < Speed.Zero)
+            {
+                throw new ArgumentOutOfRangeException(nameof(windSpeed), "The wind speed cannot be negative");
+            }
+
+            double va = temperature.DegreesCelsius;
+            if (windSpeed < Speed.FromKilometersPerHour(1))
+            {
+                // otherwise, the result is unusable, because the second and third terms of the equation are 0, resulting in a constant offset from the input temperature
+                windSpeed = Speed.FromKilometersPerHour(1);
```

Comment:
would it make sense to just return temperature here instead?

Reply:
No, because if the wind speed is low, the returned temperature is above the input temperature. This is expected and explained in the documentation. Only if the wind speed is zero or almost zero, the formula gets completely off. So this condition basically limits the output temperature to be at most about 2 degrees over the input temperature.